src/rdf4cpp/BigDecimal.hpp

```diff
@@ -250,22 +190,50 @@ struct BigDecimal {
         } else if (ex < 0) {
             exponent = static_cast<Exponent_t>(-ex);
             UnscaledValue_t e{0};
-            if (pow_checked<OverflowMode::Checked>(UnscaledValue_t{5}, -ex, e))
+            if (util::detail::pow_checked<OverflowMode::Checked>(UnscaledValue_t{5}, -ex, e))
                 throw std::overflow_error{exc};
-            if (mul_checked<OverflowMode::Checked>(UnscaledValue_t{significand}, e, unscaled_value))
+            if (util::detail::mul_checked<OverflowMode::Checked>(UnscaledValue_t{significand}, e, unscaled_value))
                 throw std::overflow_error{exc};
         } else {
             UnscaledValue_t e{0};
-            if (pow_checked<OverflowMode::Checked>(UnscaledValue_t{2}, ex, e))
+            if (util::detail::pow_checked<OverflowMode::Checked>(UnscaledValue_t{2}, ex, e))
                 throw std::overflow_error{exc};
-            if (mul_checked<OverflowMode::Checked>(UnscaledValue_t{significand}, e, unscaled_value))
+            if (util::detail::mul_checked<OverflowMode::Checked>(UnscaledValue_t{significand}, e, unscaled_value))
                 throw std::overflow_error{exc};
         }
         if (neg)
             unscaled_value = -unscaled_value;
         normalize();
     }
 
+public:
+    /**
+     * converts a double to a BigDecimal.
+     * this conversion might not be exact, due to the built in limitations of doubles.
+     * if you have the possibility, use one of the other constructors.
+     * @param value
+     * @throw std::overflow_error on exceeding the types numeric limits
+     */
+    explicit BigDecimal(double value) : unscaled_value(0), exponent(0) {
+        if constexpr (!std::numeric_limits<UnscaledValue_t>::is_bounded) {
```

Comment:
tbh, I would just remove the template parameters if you need to add special cases for different instantiations